pangeo_forge_recipes/transforms.py

```diff
@@ -638,3 +642,100 @@ def expand(
         # consolidate metadata and/or coordinate dims here
 
         return singleton_target_store
+
+
+@dataclass
+class CreatePyramid(beam.PTransform):
+    level: int
+    epsg_code: Optional[str] = None
+    rename_spatial_dims: Optional[dict] = None
+    pyramid_kwargs: Optional[dict] = field(default_factory=dict)
+
+    def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
+        return pcoll | beam.Map(
+            create_pyramid,
+            level=self.level,
+            epsg_code=self.epsg_code,
+            rename_spatial_dims=self.rename_spatial_dims,
+            pyramid_kwargs=self.pyramid_kwargs,
+        )
+
+
+@dataclass
+class StoreToPyramid(beam.PTransform, ZarrWriterMixin):
+
+    # Open design question. Should StoreToPyramid repeat all
+    # of the StoreToZarr kwargs and pass them through to the nested StoreToZarr
+    # or should it packaged them as store_to_zarr_kwargs and unpack?
+
+    """Store a PCollection of Xarray datasets as a Zarr Pyramid.
+    :param n_levels: Number of pyramid levels to generate
```

Comment:
```suggestion
    :param levels: Number of pyramid levels to generate
```
I think we should name this levels so that it's extensible for https://github.com/carbonplan/ndpyramid/issues/74